provider-anynines/internal/controller/servicebinding/servicebinding.go

```diff
@@ -262,12 +263,44 @@ func (c external) GetServiceInstanceManagedResource(ctx context.Context, sb v1.S
 	return instances.ToServiceInstance(sb.Name)
 }
 
-// initializeServiceBindingStatus initializes InstanceID, ServiceID and PlanID in the status
-// if not set.
+// GetServiceBindingSecret retrieves the servicebinding secret with postfix '-creds'.
+func (c external) GetServiceBindingSecret(ctx context.Context, sb v1.ServiceBinding) (*corev1.Secret, error) {
+	secret := &corev1.Secret{}
+
+	err := c.kube.Get(ctx, types.NamespacedName{
+		Name:      sb.Name + "-creds",
```

Comment:
This code will never retrieve the service binding secret because it uses the managed resource name instead of the claim name.
```suggestion
		Name:      sb.Labels[constants.LabelKeyClaimName] + "-creds",
```